app/src/main/java/com/dgaf/happyhour/Adapter/RestaurantDealAdapter.java

```diff
@@ -0,0 +1,213 @@
+package com.dgaf.happyhour.Adapter;
+
+import android.app.Activity;
+import android.util.Log;
+import android.widget.CompoundButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.widget.ToggleButton;
+
+import com.dgaf.happyhour.Device;
+import com.dgaf.happyhour.Model.AvailabilityModel;
+import com.dgaf.happyhour.Model.DealIcon;
+import com.dgaf.happyhour.Model.DealModel;
+import com.dgaf.happyhour.Model.ModelUpdater;
+import com.dgaf.happyhour.Model.Queries.QueryParameters;
+import com.dgaf.happyhour.R;
+import com.parse.FunctionCallback;
+import com.parse.ParseCloud;
+import com.parse.ParseException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by Adam on 8/26/2015.
+ */
+public class RestaurantDealAdapter implements ModelUpdater<DealModel> {
+    private Activity activity;
+    private DealModel dealModel;
+    private QueryParameters queryParams;
+    private RestaurantDealViewHolder dealHolder;
+
+    public static class RestaurantDealViewHolder implements CompoundButton.OnCheckedChangeListener {
+        public TextView dealTitle;
```

Comment:
I'm unsure why all these TextViews are `public`, but from a design perspective, we want to promote loose coupling